### ISSUE TITLE
Exception in SettingsEditor while configuring beacons

### ISSUE THREAD

**Issue by puja1908** [Estimote/Android-SDK#209]
Hi !! From this doc [(https://github.com/Estimote/Android-SDK/blob/master/Docs/DOC_deviceConnection.md#connecting-to-the-connectionprovider-service)] , I have implemented the whole Scanner->DeviceConnection->SettingsEditor flow to connect to and configure each of my beacons.
Basically what happens is, in onConnected success callback, when I call the function where I have to edit the settings, this exception gets thrown. I don't understand, the device is being edited only after a successful connection!

java.lang.NullPointerException: Protocol not yet detected - cannot edit device settings. You need to successfully connect to device first.

04-12 11:49:05.603 E/AndroidRuntime(31812): at com.estimote.coresdk.common.internal.utils.Preconditions.checkNotNull(Preconditions.java:21)

04-12 11:49:05.603 E/AndroidRuntime(31812): at com.estimote.mgmtsdk.connection.strategy.DeviceConnectionInternal.getCloudSyncEngine(DeviceConnectionInternal.java:157)

**Comment by puja1908** [comment]
If it helps, 
Beacons firmware version - 4.9.4 and,
Test Devices - Samsung S5- 6.0, Samsung S5 - 5.1.1.

**Comment by puja1908** [comment]
E/EstimoteSDK: DeviceCloudSyncEngine$3$1.failure:165 Device status update failed
                                                          com.estimote.coresdk.common.exception.EstimoteCloudException: 400 Bad Request
                                                              at com.estimote.coresdk.cloud.internal.InternalEstimoteCloud$7.failure(InternalEstimoteCloud.java:511)
                                                              at com.estimote.coresdk.repackaged.retrofit_v1_9_0.retrofit.CallbackRunnable$2.run(CallbackRunnable.java:53)
                                                              at android.os.Handler.handleCallback(Handler.java:739)
                                                              at android.os.Handler.dispatchMessage(Handler.java:95)
                                                              at android.os.Looper.loop(Looper.java:148)
                                                              at android.app.ActivityThread.main(ActivityThread.java:7329)
                                                              at java.lang.reflect.Method.invoke(Native Method)
                                                              at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
                                                              at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)

Since yesterday, I'm facing this new issue..after connecting to the beacon, the writeSettings fail with this exception...This is urgent... Please help !!

**Comment by pawelDylag** [comment]
Hello @puja1908 

Did you initialised our SDK with the proper AppKey and AppToken?
Could you also confirm that this is still occurring in sdk `version 1.0.2` ?
Which settings are you trying to update via `SettingsEditor`?

**Comment by puja1908** [comment]
Hi @pawelDylag 

I have resolved these issues with some logic changes from my side, but to still answer your queries, 
1. I have initialized the SDK with the proper AppKey and AppToken. 
2. I have tested on 1.0.1 and it's working fine for now.
3. I'm updating the UUID and Major.

For anyone else who runs into the same exceptions, please feel free to ping me. Will be glad to help you out! :)

**Comment by pawelDylag** [comment]
Hey @puja1908 

Cool! If you have any more problems or ideas how to make such operations better, let me know!

Cheers!

**Comment by CrossPT** [comment]
@pujaagrawal1 i'm currently having the same problem? How did you fix it?
